assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image/edit.tsx

```diff
@@ -35,8 +41,33 @@ export const Edit = (): JSX.Element => {
 			templateInsertUpdatesSelection: true,
 		}
 	);
+
+	const previousNextImage = useMemo( () => {
```

Comment:
Good point. I have followed the original code from: https://github.com/woocommerce/woocommerce-blocks/blob/b2390ece0c1606a25b84997cc0cf8ac7780c8959/assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image-next-previous/edit.tsx#L46

@gigitux what do you think?

Reply:
I don't have a strong opinion, but if we can avoid doing this check at each re-render, why not?

I suggest to read from react documentation: https://react.dev/reference/react/useMemo#:~:text=There%20is%20no,an%20entire%20component.

Reply:
Just an FYI more than anything [here's another reference](https://kentcdodds.com/blog/usememo-and-usecallback#whats-the-point)

>Performance optimizations are not free. They ALWAYS come with a cost but do NOT always come with a benefit to offset that cost. Therefore, optimize responsibly.

Reply:
> Just an FYI more than anything [here's another reference](https://kentcdodds.com/blog/usememo-and-usecallback#whats-the-point)

TIL

Since we invoke `setAttributes` for each settings change, the block could be re-render many times. As I said, I don't have a strong preference. To have a "real" answer we should benchmark and see which solution is better, but in this case, it isn't worth it. Since it is a micro-optimization, it is okay for me both approaches.